express-server/middleware/errorHandler.js

```diff
@@ -0,0 +1,19 @@
+const { ServerError } = require('../errors/ServerError');
+const { logger } = require('../logger/logger.js');
+const { HTTPStatuses } = require('../constants/HTTPStatus.js');
+
+exports.wrapError = (fn) => (req, res, next) => {
+    Promise.resolve(fn(req, res, next)).catch(next)
+};
+
+exports.errorMiddleware = (err, req, res, next) => {
+  if (err instanceof ServerError) {
+    res.status(err.HTTPStatus).send(err.message);
+    logger.error(`custom server error: ${err.message}`);
+  } else {
+    res
+      .status(HTTPStatuses.INTERNAL_SERVER_ERROR)
+      .send('internal server error');
```

Comment:
you're masking the error from the client. on one hand this has advantages, the client doesn't need to know what exactly failed. on the other hand it has a lot of cons:
* hard to debug
* hard to monitor
* all of the errors looks the same, the client doesn't know why it failed

Reply:
i mask the error when i have some internal server error that i didn't handle, what would you advise me to do in this case? i.e. lets take a scenario where i try to access "person.age" but person does not exists, so an error will be thrown and in this case, an INTERNAL SERVER ERROR will be returned to the user, what should i return instead in this case?

Reply:
for this project I would return the error as is. for real projects I would send the client a useful link to follow the reason of the error. for example in wix we always return the client the request id. Then we can trace the error in new relic / grafana